File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int[] arr) {
        Stack<Integer> stack = new Stack<>();
        ArrayList<Integer> list = new ArrayList<>();
        
        for (int i = 0; i<arr.length; i++) {
            if (stack.isEmpty()) {
                stack.add(arr[i]);
            } else {
                if (stack.peek() != arr[i]) {
                    list.add(stack.pop());
                    stack.add(arr[i]);
                }
            }
        }
        if (!stack.isEmpty()) {
            list.add(stack.pop());
        }
        
        int[] answer = list.stream()
                .mapToInt(Integer::intValue)
                .toArray();
        return answer;
    }
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    answer.append(arr[0])
    if (len(arr) == 1):
        return answer

    for i in range(1, len(arr)):
        if answer[len(answer)-1] != arr[i]:
            answer.append(arr[i])
    return answer